### PR TITLE
507 marketplace e2e test setup

### PR DESCRIPTION
**Pull request by gjermundgaraba**

### CODE REVIEW

File: chain/tests/e2e/plasticcredit/applicant_tx_test.go [hunk 1]
```diff
@@ -178,18 +179,18 @@ func (s *E2ETestSuite) TestCmdUpdateApplicant() {
...
 				s.Require().Contains(out.String(), tc.expectedErrMsg)
 			case tc.expectedErrOnExec:
-				txResponse, err := s.getCliResponse(val.ClientCtx, out.Bytes())
+				txResponse, err := s.GetCliResponse(val.ClientCtx, out.Bytes())
 				s.Require().NoError(err)
 				s.Require().NotEqual(uint32(0), txResponse.Code)
 				s.Require().Contains(txResponse.RawLog, tc.expectedErrMsg)
 			default:
-				cliResponse, err := s.getCliResponse(val.ClientCtx, out.Bytes())
+				cliResponse, err := s.GetCliResponse(val.ClientCtx, out.Bytes())
```
Comment (maheshKRex) [inline chain/tests/e2e/plasticcredit/applicant_tx_test.go:193]: @gjermundgaraba does these updates meaning, we are updating getCliResponse with some other function GetCliResponse, just couldn't find the implementation for such method with 'G' in caps.

Comment (gjermundgaraba) [reply chain/tests/e2e/plasticcredit/applicant_tx_test.go:193]: I moved the base part of the E2ETestSuite into a separate package so it was reusable. When I did that I needed to make this (and a few other things) public. In go capital letter is what makes it public.

So it's the same function, just been moved to a different package and therefore needs to be public.

**Review by maheshKRex — APPROVED** [review]

**Review by ba1ciu — APPROVED** [review]
Looks nice! Just one design question regarding test setup in a comment.

File: chain/tests/e2e/marketplace/e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+const wasmFilePath = "../../../../cosmwasm/artifacts/plastic_credit_marketplace.wasm"
+
+type E2ETestSuite struct {
+	e2e.TestSuite
+}
+
+// uploadContractBeforeAll uploads the marketplace contract before all tests
+// because of this, we have the code id 1 set already and can use it in other tests
+func uploadContractBeforeAll(s *e2e.TestSuite) {
```
[the remark this answers is not in the document]
Comment (gjermundgaraba) [reply chain/tests/e2e/marketplace/e2e_test.go:36]: It depends. The goal here was to upload a contract only for the marketplace tests, not for the plastic credits e2e tests. They run separately so they can't interfere with each others state.

**Pull request merged**